subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -190,10 +189,6 @@ private String getFilename(Path dir) {
         public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
             String internedName = intern(file.getFileName().toString());
             if (shouldVisit(file, internedName, false, attrs, builder.getRelativePath())) {
-                if (attrs.isSymbolicLink()) {
```

Comment:
Should we return the missing file snapshot directly here, without waiting for the hashing to fail? I think this would be better in the long run, given that we want to handle unavailable files and broken symlinks slightly different.

Reply:
It doesn't actually get to hashing, the [regular file check](https://github.com/gradle/gradle/blob/a4d285eb3b8006b31b4699aacbde281d99776d60/subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java#L199) should filter it out.